src/components/Notification.tsx

```diff
@@ -0,0 +1,36 @@
+import { FC } from 'react';
+import classNames from 'classnames';
+import { NotificationType } from '../types/NotificationType';
+import { useGlobalContext } from '../hooks/useGlobalContext';
+
+type Props = {
+  type: NotificationType;
+} & (
+  {
+    type: NotificationType.PostsWarning;
+    warningText: string
+  } | {
+    type: Exclude<NotificationType, NotificationType.PostsWarning>;
+    warningText?: string;
+  }
+);
```

Comment:
as for me, you can remove this first part 
```suggestion
type Props =
  {
    type: NotificationType.PostsWarning;
    warningText: string
  } | {
    type: Exclude<NotificationType, NotificationType.PostsWarning>;
    warningText?: string;
  }
```